pkg/rpc/context.go

```diff
@@ -107,52 +108,118 @@ type connMeta struct {
 	heartbeatErr error
 }
 
-// Context contains the fields required by the rpc framework.
-type Context struct {
+// ContextTestingKnobs are the testing knobs for a Context.
+type ContextTestingKnobs struct {
+	BreakerFactory func() *circuit.Breaker
+	// PeriodicClockCheckHook is used to override the default clock offset check
+	// on connections. If an error is returned, the node will crash itself.
+	PeriodicClockCheckHook func() error
+}
+
+// ModuleTestingKnobs is part of the base.ModuleTestingKnobs interface.
+func (*ContextTestingKnobs) ModuleTestingKnobs() {
+}
+
+var _ base.ModuleTestingKnobs = &ContextTestingKnobs{}
+
+// ContextConfig contains parameters for creating a Context.
+type ContextConfig struct {
+	// The base.Config used to open gRPC connections.
 	*base.Config
 
+	// KeepaliveInterval is the interval at which gRPC is going to perform HTTP2
+	// pings to check that the connection is still alive.
+	KeepaliveInterval time.Duration
+	// KeepaliveTimeout is the timeout for receiving a response to a ping. If no
+	// response is received within timeout, the connection is closed and in-flight
+	// RPCs will receive an error. gRPC will attempt to reconnect.
+	KeepaliveTimeout time.Duration
+
+	// HeartbeatInterval is the period on which the health status of a connection
```

Comment:
"health status of a connection" suggests a lot of overlap with the keepalive. Maybe more accurate to say "the health status of a remote node" instead?